safe.py

```diff
@@ -735,6 +735,14 @@ def __repr__(self):
   def copy(self):
     # Create a new instance
     copy_inst = SafeDict(self.__under__)
+    
+    # Fix for recursion depth reached issue #97 when copying and printing a   
+    # SafeDict containing a reference to itself.
+    # Caveat: dict.copy is expected to return a shallow copy, this fix 
+    # introduces a partial deep copy for the contained self reference
+    for key, value in self.__under__.iteritems():
+      if value is self:
+        copy_inst[key] = copy_inst
```

Comment:
Is `_context` expected to be the only dict key to have a reference to itself?  Why not check explicitly for the `_context` key, instead of looping through the entire SafeDict?